react-graph/src/PointNode.tsx

```diff
@@ -32,48 +32,279 @@ export class PointNode {
+            this.AggregateChildStats();
+        }
+    }
+
+    /**
+     * Updates statistics based on the current points.
+     */
+    private CalculatePointStats(): void {
+        if (this.points === null || this.points.length === 0) return;
+
+        this.count = this.points.length;
+        this.minT = this.points?.[0]?.[0] ?? NaN;
+        this.maxT = this.points?.[this.points.length - 1]?.[0] ?? NaN;
+        this.dim = this.points?.[0]?.length ?? NaN;
+
+        for (let index = 1; index < this.dim; index++) {
+            const values = this.points.map(pt => pt[index]);
+            this.minV[index - 1] = Math.min(...values);
+            this.maxV[index - 1] = Math.max(...values);
+            this.sum[index - 1] = values.reduce((acc, val) => acc + val, 0);
         }
-        for (let index = 0; index < this.dim - 1; index++) this.minV[index] = Math.min(...this.children.map(node => node.minV[index]));
-        for (let index = 0; index < this.dim - 1; index++) this.maxV[index] = Math.max(...this.children.map(node => node.maxV[index]));
     }
 
-    public GetData(Tstart: number, Tend: number, IncludeEdges?: boolean): [...number[]][] {
+    /**
+     * Updates statistics based on the current children.
+     */
+    private AggregateChildStats(): void {
+        if (this.children === null || this.children.length === 0) return;
+
+        this.minT = Math.min(...this.children.map(node => node.minT));
+        this.maxT = Math.max(...this.children.map(node => node.maxT));
+
+        for (let index = 0; index < this.dim - 1; index++) {
+            this.minV[index] = Math.min(...this.children.map(node => node.minV[index]));
+            this.maxV[index] = Math.max(...this.children.map(node => node.maxV[index]));
+            this.sum[index] = this.children.reduce((s, node) => s + node.sum[index], 0);
+        }
+        this.count = this.children.reduce((acc, node) => acc + node.count, 0);
+    }
+
+    /**
+     * Updates aggregated statistics for this node to include a newly added child node.
+     * 
+     * @param {PointNode} newChild - The new child node whose statistics will be merged into this node.
+     */
+    private IncrementStatsForNewChild(newChild: PointNode): void {
+        // Update the time range
+        this.minT = Math.min(this.minT, newChild.minT);
+        this.maxT = Math.max(this.maxT, newChild.maxT);
+
+        // Update value ranges and sums
+        for (let i = 0; i < this.dim - 1; i++) {
+            this.minV[i] = Math.min(this.minV[i], newChild.minV[i]);
+            this.maxV[i] = Math.max(this.maxV[i], newChild.maxV[i]);
+            this.sum[i] += newChild.sum[i];
+        }
+
+        this.count += newChild.count;
+    }
+
+    /**
+     * Updates aggregated statistics for this node to include a newly added point.
+     * 
+     * @param {number[]} newPt - The new point
+     */
+    private IncrementStatsForNewPoint(newPt: number[]): void {
+        this.count += 1;
```

Comment:
This may cause an issue with the constructor already setting it to `1` can you double check that?